### PR TITLE
Add Astra DB (/Cassandra) to vector databases with example notebooks

### PR DESCRIPTION
**Pull request by hemidactylus**
This PR adds a folder about using the Vector Search capabilities in Astra DB / Apache Cassandra® with OpenAI
(specifically, vector embeddings and chat completion).

The notebooks guide through a standard exercise (retrieval and then text generation over a textual source labeled dataset).

There is a README acting as index to the folder contents.

### Note to test the notebooks

To anyone who might want to give a try to the colab notebooks: the links in this PR _assume the code live in the `main` branch of `openai`'s repo, which of course is currently not the case yet. So here I provide the "open in colab" URLs that work right now:

- [using cassIO library](https://colab.research.google.com/github/hemidactylus/openai-cookbook/blob/SL-cassandra_astra_vector/examples/vector_databases/cassandra_astradb/Philosophical_Quotes_cassIO.ipynb)
- [using direct CQL](https://colab.research.google.com/github/hemidactylus/openai-cookbook/blob/SL-cassandra_astra_vector/examples/vector_databases/cassandra_astradb/Philosophical_Quotes_CQL.ipynb)

Additionally, for the same reason, testing the Colab notebooks would require replacing the definition of `json_url` in both notebooks with the following:

```
json_url = "https://raw.githubusercontent.com/hemidactylus/openai-cookbook/SL-cassandra_astra_vector/examples/vector_databases/cassandra_astradb/sources/philo_quotes.json"
```

### CODE REVIEW

**Comment by derrickCos** [comment]
Can we get a review on this? this would be awesome for Cassandra community

File: examples/vector_databases/cassandra_astradb/Philosophical_Quotes_CQL.ipynb [hunk 1]
```diff
@@ -0,0 +1,1353 @@
+{
```
Comment (simonpfish) [inline examples/vector_databases/cassandra_astradb/Philosophical_Quotes_CQL.ipynb:1]: <p>minor typo: <code>This notebook can run either as a Colab notebook or <s>with</s> as a regular Jupyter notebook.</code> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/openai/openai-cookbook/pull/655/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='1'/>

Comment (simonpfish) [inline examples/vector_databases/cassandra_astradb/Philosophical_Quotes_CQL.ipynb:1]: <p>Here we could reference and link to https://github.com/openai/openai-cookbook/blob/main/examples/Question_answering_using_embeddings.ipynb as a general guide for this </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/openai/openai-cookbook/pull/655/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='2'/>

File: examples/vector_databases/cassandra_astradb/Philosophical_Quotes_cassIO.ipynb [hunk 2]
```diff
@@ -0,0 +1,1161 @@
+{
```
Comment (simonpfish) [inline examples/vector_databases/cassandra_astradb/Philosophical_Quotes_cassIO.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #3.    generation_prompt_template = """"Generate a single short philosophical on the given topic,
</pre><p>there might be a typo here, i think you mean to write "short philosophical <strong>quote </strong>on..."</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/openai/openai-cookbook/pull/655/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='44'/>

Comment (simonpfish) [inline examples/vector_databases/cassandra_astradb/Philosophical_Quotes_cassIO.ipynb:1]: <p>This first sentence could perhaps be improved a bit for clarity, maybe something along the lines of: </p><p> </p><blockquote>Congratulations! You have learned how to use OpenAI for vector embeddings and Astra DB / Cassandra for storage in order to build a sophisticated philosophical search engine and quote generator.</blockquote>
<br/>

 _Reply via <a href='https://app.reviewnb.com/openai/openai-cookbook/pull/655/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='66'/>

**Review by simonpfish — APPROVED** [review]
Thank you for the high quality examples. Overall everything looks good, left a couple comments here and there for language improvements. Feel free to merge once those are resolved.

**Review by whoami-anoint — APPROVED** [review]

**Comment by hemidactylus** [comment]
Thank you for the valid remarks and suggested improvements. I implemented them all (in both notebooks where applicable).
This should be now good to go.

**Pull request merged**